src/controllers/commands_controller.js

```diff
@@ -32,153 +32,120 @@
-              return;
-            }
-            const iotUrl = CommandsController.gatewayHref + payload.href;
-            iotOptions.headers.Authorization = request.headers.authorization;
-            // Returning 201 to signify that the command was mapped to an
-            // intent and matched a 'thing' in our list. Return a response to
-            // caller with this status before the command finishes execution
-            // as the execution can take some time (e.g. blinds)
-            response.status(201).json({
-              message: 'Command Created',
-              payload: payload,
-            });
-            fetch(iotUrl, iotOptions)
-              .then(function() {
-                // In the future we may want to use WS to give a status of
-                // the disposition of the command execution..
-              })
-              .catch(function(err) {
-                // Future, give status via WS.
-                console.log(`catch inside PUT:${err}`);
-              });
-          } else {
-            response.status(404).json({message: 'Thing not found'});
-          }
-        }).catch(function(error) {
-          console.log('Error parsing intent:', error);
-          response.status(404).json({message:
-            'Internal error determining intent'});
-        });
-      }).catch(function(error) {
-        console.log('Error training:', error);
-        response.status(404).json({message:
-          'Internal error determining intent'});
-      });
-    }).catch(function(err) {
-      console.log(`error catch:${err}`);
-      response.status(500).send({
-        message: err,
-      });
+  let names = await Things.getThingNames();
+  names = names.map((n) => n.toLowerCase());
+
+  try {
+    await IntentParser.train(names);
+  } catch (e) {
+    console.log('Error training:', e);
+    response.status(400).json({message: 'Internal error determining intent'});
+    return;
+  }
+
+  let payload;
+  try {
+    payload = await IntentParser.query(request.body.text);
+  } catch (e) {
+    console.log('Error parsing intent:', e);
+    response.status(400).json({message: 'Internal error determining intent'});
+    return;
+  }
+
+  const name = payload.thing;
+  const thing = await Things.getThingByName(name);
+
+  if (thing) {
+    let propertyName, value;
+
+    if (payload.value === 'on' || payload.value === 'off') {
+      propertyName = 'on';
+      value = payload.value === 'on';
+    } else if ((payload.value === 'warmer' ||
+                payload.value === 'cooler') &&
+               thing.properties.colorTemperature) {
+      let current;
+      try {
+        current = await AddonManager.getProperty(thing.id, 'colorTemperature');
+      } catch (e) {
+        response.status(400).json({message: 'Failed to set property'});
+        return;
+      }
+
+      propertyName = 'colorTemperature';
+      value = payload.value === 'warmer' ? current - 100 : current + 100;
```

Comment:
These adjustment values may need to be tweaked a bit, not sure what others think.